ui/src/components/Secrets.vue

```diff
@@ -517,6 +581,58 @@ function removeSecret(counter) {
     secrets.value[0] = { key: "", value: "", file: [] }
   }
 }
+
+function generateSecureRandomString(length, includeUppercase, includeLowercase, includeNumbers, includeSpecial) {
+  const chars = [];
+  const result = [];
+  if (includeUppercase) {
+    chars.push(...'ABCDEFGHIJKLMNOPQRSTUVWXYZ');
+    result.push(getRandomCharacter('ABCDEFGHIJKLMNOPQRSTUVWXYZ'));
+  }
+  if (includeLowercase) {
+    chars.push(...'abcdefghijklmnopqrstuvwxyz');
+    result.push(getRandomCharacter('abcdefghijklmnopqrstuvwxyz'));
+  }
+  if (includeNumbers) {
+    chars.push(...'0123456789');
+    result.push(getRandomCharacter('0123456789'));
+  }
+  if (includeSpecial) {
+    chars.push(...'@#$%&*()');
+    result.push(getRandomCharacter('@#$%&*()'));
+  }
+
+  if (chars.length === 0) {
+    setErrorMessage('At least one character type must be selected to create a random string.');
+    return '';
+  }
+  for (let i = result.length; i < length; i++) {
+    result.push(getRandomCharacter(chars.join('')));
+  }
+  shuffleArray(result);
+  return result.join('');
+}
+
+function getRandomCharacter(charSet) {
+  const randomValues = new Uint8Array(1);
+  window.crypto.getRandomValues(randomValues);
+  const randomIndex = randomValues[0] % charSet.length;
```

Comment:
I am not sure, if this would favour some chars over others, depending on the length of charSet and the bit-size of the random number.